### PR TITLE
[WIP] trying pet ui

### PR DESCRIPTION
**Pull request by Al-Andrew**

### CODE REVIEW

File: src/DucksNet.Domain/Model/Pet.cs [hunk 1]
```diff
@@ -1,54 +1,104 @@
...
+        Name = name;
+        DateOfBirth = dateOfBirth;
+        Species = species;
+        Breed = breed;
         OwnerId = ownerId;
         Size = size;
     }
 
-    private Pet(Size size)
+    private Pet(string name, DateTime dateOfBirth, string species, string breed, Guid ownerId, Size size) // TODO (Ad): This constructor only exists to make it json parsable
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/Pet.cs]: remove the comment. this constructor is private

File: src/DucksNet.Domain/Model/Pet.cs [hunk 2]
```diff
@@ -1,54 +1,104 @@
+﻿using System;
+using System.Globalization;
 using DucksNet.Domain.Model.Enums;
 using DucksNet.SharedKernel.Utils;
+using static System.Runtime.InteropServices.JavaScript.JSType;
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/Pet.cs:6]: do you need these three? (line 1, 2 and 5)

File: src/DucksNet.Domain/Model/User.cs [hunk 3]
```diff
@@ -1,48 +1,94 @@
...
+        {
             return Result<User>.Error("Password is required.");
-
-        return Result<User>.Ok(new User(name, email, password));
+        }
+        if (!Validation.IsTelephoneNumberValid(phoneNumber))
+        {
+            return Result<User>.Error("The phone number is not valid.");
+        }
+        if (!Validation.IsEmailValid(email))
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/User.cs:50]: resolve the warnings

File: src/DucksNet.Domain/Model/User.cs [hunk 4]
```diff
@@ -1,48 +1,94 @@
...
-        if (!string.IsNullOrWhiteSpace(name))
+        if (!string.IsNullOrWhiteSpace(firstName))
+        {
+            FirstName = firstName;
+        }
+        if (!string.IsNullOrWhiteSpace(lastName))
+        {
+            LastName = lastName;
+        }
+        if (Validation.IsTelephoneNumberValid(phoneNumber))
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/User.cs:67]: resolve the warnings

File: src/DucksNet.Domain/Model/User.cs [hunk 5]
```diff
@@ -1,48 +1,94 @@
...
+        {
+            PhoneNumber = phoneNumber;
+        }
+        if (!string.IsNullOrWhiteSpace(address))
         {
-            Name = name;
+            Address = address;
         }
-        if (!string.IsNullOrWhiteSpace(email))
+        if (Validation.IsEmailValid(email))
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/User.cs:75]: resolve the warnings

File: tests/DucksNet.IntegrationTests/CagesControllerTests.cs [hunk 6]
```diff
@@ -1,4 +1,4 @@
-using System;
+﻿using System;
```
Comment (OpariucRares) [inline tests/DucksNet.IntegrationTests/CagesControllerTests.cs:1]: do you need this?

File: tests/DucksNet.IntegrationTests/PetControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (OpariucRares) [inline tests/DucksNet.IntegrationTests/PetControllerTests.cs]: what is this? don't add empty files. Add tests

File: tests/DucksNet.UnitTests/PetTests.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+namespace DucksNet.UnitTests;
+public class PetTests
+{
+    [Fact]
+    public void When_CreatePet_Then_ShouldReturnPet()
+    {
+        Tuple<string, DateTime, string, string, Guid, string> sut = CreateSUT();
+        var result = Pet.Create(sut.Item1, sut.Item2, sut.Item3, sut.Item4, sut.Item5, sut.Item6);
+        var copy = result.Value;
+        copy.Name.Should().Be(sut.Item1);
```
Comment (OpariucRares) [inline tests/DucksNet.UnitTests/PetTests.cs]: resolve the warning

File: tests/DucksNet.UnitTests/UserTests.cs [hunk 9]
```diff
@@ -0,0 +1,76 @@
+﻿using DucksNet.Domain.Model;
+using System;
+
+namespace DucksNet.UnitTests;
+public class UserTests
+{
+    //// TODO (Ad): see why id is null and the test fails
+    //[Fact]
+    //public void When_CreateUser_Then_ShouldReturnUser()
```
Comment (OpariucRares) [inline tests/DucksNet.UnitTests/UserTests.cs]: why is this commented?

**Review by OpariucRares — CHANGES_REQUESTED** [review]
don't add the DucksNet.db, DucksNet.db-shm and DucksNet.db-wal and resolve the Code smells (if it is possible)

File: tests/DucksNet.UnitTests/UserTests.cs (hunk 9 above)
Comment (adinaned) [reply tests/DucksNet.UnitTests/UserTests.cs]: Pica testul si nu am mai avut timp sa vad de ce

File: tests/DucksNet.IntegrationTests/CagesControllerTests.cs (hunk 6 above)
Comment (adinaned) [reply tests/DucksNet.IntegrationTests/CagesControllerTests.cs:1]: Da, altfel da erori

**Comment by Al-Andrew** [comment]
Also don't forget to merge with dev to get the updated tests. And add your tests to the runner otherwise you won't get coverage or testing results in CI.

File: src/DucksNet.Domain/Model/User.cs [hunk 10]
```diff
@@ -1,48 +1,95 @@
...
 
-    public static Result<User> Create(string? name, string? email, string? password)
+    private User(string firstName, string lastName, string address, string phoneNumber, string email, string password) 
     {
-        if (string.IsNullOrWhiteSpace(name))
-            return Result<User>.Error("Name is required.");
-        if (string.IsNullOrWhiteSpace(email))
-            return Result<User>.Error("Email is required.");
+        Id = Guid.NewGuid();
+        Console.WriteLine($"Create user id: {Id}");
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/User.cs:19]: We don't need this print anymore.

File: tests/DucksNet.IntegrationTests/PetsControllerTests.cs [hunk 11]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (Al-Andrew) [inline tests/DucksNet.IntegrationTests/PetsControllerTests.cs]: Do we have 0 tests for this controller? Then why do we have the file?

**Review by Al-Andrew — COMMENTED** [review]
I also don't understand why there's a drop in coverage.

**Comment by Al-Andrew** [comment]
> I also don't understand why there's a drop in coverage.

Found out. You're failing a bunch of tests. Please run all tests locally before pushing. For now the CI won't notify you if you fail tests.

File: src/DucksNet.Domain/Model/Pet.cs (hunk 1 above)
Comment (OpariucRares) [reply src/DucksNet.Domain/Model/Pet.cs]: ^

File: src/DucksNet.Domain/Model/User.cs (hunk 10 above)
Comment (OpariucRares) [reply src/DucksNet.Domain/Model/User.cs:19]: ^

File: src/DucksNet.WebUI/Pages/UsersOverview.razor.cs [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace DucksNet.WebUI.Pages;
+public partial class UsersOverview : ComponentBase
+{
+    [Inject]
+    public IUserDataService? UserDataService { get; set; }
+    public List<User> Users { get; set; } = default!;
+    protected async override Task OnInitializedAsync()
+    {
+        Users = (await UserDataService.GetAllUsers()).ToList();
```
Comment (OpariucRares) [inline src/DucksNet.WebUI/Pages/UsersOverview.razor.cs]: resolve the warning

File: tests/DucksNet.IntegrationTests/PetsControllerTests.cs (hunk 11 above)
Comment (OpariucRares) [reply tests/DucksNet.IntegrationTests/PetsControllerTests.cs]: ^

**Review by OpariucRares — CHANGES_REQUESTED** [review]
check the comments before the review. Also, resolve the code smells

**Review by OpariucRares — APPROVED** [review]
good

**Review by OpariucRares — CHANGES_REQUESTED** [review]
error tests at medical Record, Appointment and CageSchedule

**Review by OpariucRares — APPROVED** [review]
for now approve

**Pull request merged**